CMakeLists.txt

```diff
@@ -115,6 +120,11 @@ if(${use_openssl})
     if (NOT TARGET OpenSSL::SSL OR NOT TARGET OpenSSL::Crypto OR NOT ${OPENSSL_INCLUDE_DIR})
         find_package(OpenSSL REQUIRED)
     endif()
+
+    if (DEFINED OPENSSL_VERSION AND (${OPENSSL_VERSION} GREATER_EQUAL 3))
```

Comment:
Nit: worth adding a short comment with explanation of support for OpenSSL v1 using the v3 compat layer (would be nice to also have a link to official documentation).

Reply:
Added.

    # The block below enables the v1 back-compatibility layer in OpenSSL 3,
    # if using that version or later. For reference, please check the OpenSSL
    # official documentation: https://www.openssl.org/docs/man3.0/man7/openssl_user_macros.html